bundles/org.openhab.binding.anthem/src/main/java/org/openhab/binding/anthem/internal/handler/AnthemHandler.java

```diff
@@ -0,0 +1,440 @@
+    }
+
+    public void setMacAddress(String mac) {
+        updateProperty("Mac Address", mac);
+    }
+
+    public void updateChannelState(String zone, String channelId, State state) {
+        updateState(zone + "#" + channelId, state);
+    }
+
+    public void checkPowerStatusChange(String zone, String power) {
+        // Zone 1
+        if (Zone.MAIN.equals(Zone.fromValue(zone))) {
+            boolean newZone1PowerState = "1".equals(power) ? true : false;
+            if (!zone1PreviousPowerState && newZone1PowerState) {
+                // Power turned on for main zone.
+                // This will cause the main zone channel states to be updated
+                scheduler.submit(() -> queryAdditionalInformation(Zone.MAIN));
+            }
+            zone1PreviousPowerState = newZone1PowerState;
+        }
+        // Zone 2
+        else if (Zone.ZONE2.equals(Zone.fromValue(zone))) {
+            boolean newZone2PowerState = "1".equals(power) ? true : false;
+            if (!zone2PreviousPowerState && newZone2PowerState) {
+                // Power turned on for zone 2.
+                // This will cause zone 2 channel states to be updated
+                scheduler.submit(() -> queryAdditionalInformation(Zone.ZONE2));
+            }
+            zone2PreviousPowerState = newZone2PowerState;
+        }
+    }
+
+    public void setNumAvailableInputs(int numInputs) {
+        // Request the names for all the inputs
+        for (int input = 1; input <= numInputs; input++) {
+            sendCommand(AnthemCommand.queryInputShortName(input));
+            sendCommand(AnthemCommand.queryInputLongName(input));
+        }
+        updateProperty("Number of Inputs", String.valueOf(numInputs));
+    }
+
+    private void queryAdditionalInformation(Zone zone) {
+        // Request information about the device
+        sendCommand(AnthemCommand.queryNumAvailableInputs());
+        sendCommand(AnthemCommand.queryModel());
+        sendCommand(AnthemCommand.queryRegion());
+        sendCommand(AnthemCommand.querySoftwareVersion());
+        sendCommand(AnthemCommand.querySoftwareBuildDate());
+        sendCommand(AnthemCommand.queryHardwareVersion());
+        sendCommand(AnthemCommand.queryMacAddress());
+        sendCommand(AnthemCommand.queryVolume(zone));
+        sendCommand(AnthemCommand.queryMute(zone));
+        // Give some time for the input names to populate before requesting the active input
+        scheduler.schedule(() -> queryActiveInput(zone), 5L, TimeUnit.SECONDS);
+    }
+
+    private void queryActiveInput(Zone zone) {
+        sendCommand(AnthemCommand.queryActiveInput(zone));
+    }
+
+    private void sendCommand(AnthemCommand command) {
+        logger.debug("Adding command to queue: {}", command);
+        sendQueue.add(command);
+    }
+
+    private synchronized void connect() {
+        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.NONE, "@text/thing-status-detail-initializing");
```

Comment:
This should be removed, `initialize()` should set status already before calling this method.

Reply:
Ok